File: 0067-add-binary/0067-add-binary.py
class Solution(object):
    def addBinary(self, a, b):
        s=0;s1=0;h=""
        t=len(a)-1
        for i in a:
            s+=int(i)*(2**t)
            t=t-1
        t=len(b)-1
        for i in b:
            s1+=int(i)*(2**t)
            t=t-1
        s2=s1+s
        if(s2==0):
            return "0"
        while(s2>=1):
            h+=str(s2%2)
            s2=s2/2
        return h[::-1]

File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string ans = "";
        int i = a.size()-1;
        int j = b.size()-1;
        int sum=0;
        int carry=0;
        while(i>=0 || j>=0){
            if(i>=0){
                sum = sum + a[i]-'0';
            }
            if(j>=0){
                sum = sum + b[j]-'0';
            }
            if((sum+carry)==2){
                carry = 1;
                ans = "0" + ans;
            }
            else if((sum+carry)==3){
                carry = 1;
                ans = "1" + ans;
            }
            else{
                sum = sum + carry;
                if(sum){
                    ans = "1" + ans;
                    carry = 0;
                }
                else{
                    ans = "0" + ans;
                    carry = 0;
                }
            }
            sum = 0;
            i--;
            j--;
        }
        if(carry){
            ans = "1" + ans;
            return ans;
        }
        return ans;
    }
};